sim/src/main/cc/bridges/UARTModule.cc

```diff
@@ -0,0 +1,75 @@
+// See LICENSE for license details.
+
+#ifndef RTLSIM
+#include "simif_f1.h"
+#define SIMIF simif_f1_t
+#else
+#include "simif_emul.h"
+#define SIMIF simif_emul_t
+#endif
+
+#include "bridges/uart.h"
+#include "simif_peek_poke.h"
+
+class UARTModuleTest final : public simif_peek_poke_t {
+public:
+  class Handler final : public uart_handler {
+  public:
+    ~Handler() {
+      // Check that the input and output buffers are equal.
+      if (in_buffer != out_buffer) {
+        fprintf(stderr,
+                "Buffer mismatch:\n  %s\n  %s\n",
+                in_buffer.c_str(),
+                out_buffer.c_str());
+        abort();
+      }
+    }
+
+    std::optional<char> get() override {
+      if (in_index >= in_buffer.size())
+        return std::nullopt;
+      return in_buffer[in_index++];
+    }
+
+    void put(char data) override { out_buffer += data; }
+
+  private:
+    const std::string in_buffer = "We are testing the UART bridge";
+    std::string out_buffer;
+    size_t in_index = 0;
+  };
+
+  UARTModuleTest(simif_t &simif)
+      : simif_peek_poke_t(&simif, PEEKPOKEBRIDGEMODULE_0_substruct_create),
+        uart(std::make_unique<uart_t>(&simif,
+                                      UARTBRIDGEMODULE_0_substruct_create,
+                                      std::make_unique<Handler>())) {}
+
+  void run() {
+    // Reset the device.
+    poke(reset, 1);
+    step(1);
+    poke(reset, 0);
+    step(1);
+
+    // Tick for a while.
+    for (unsigned i = 0; i < 30; ++i) {
+      uart->tick();
+      step(10000);
+    }
+  }
+
+private:
+  std::unique_ptr<uart_t> uart;
+};
+
+int main(int argc, char **argv) {
```

Comment:
Should future tests in this project also define there own mains like this? I need to add some TracerV features and wanted to write tests rebased against this branch

Reply:
For now, let's define the same method everywhere. The entry points will change in subsequent PRs involving the DPI changes and we can introduce a uniform representation there.